### PR TITLE
[#9] Support API key

### PR DESCRIPTION
**Pull request by daenamkim**
Resoves #9 

- Support API key instead of login/password
- Fix all broken API call, sending transaction
- Add UI tests with mock3
- Refactor

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -2,32 +2,24 @@
...
-axios.defaults.withCredentials = true;
-
-// The local address where the demo app will be served
-const baseURL = 'https://localhost:8080/';
-
-// Configure these values with your MultiBaas login information
-const apiUser = '';
-const apiPassword = '';
+// [NEED TO FILL] API key
+const API_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1OTU4NTIzOTUsInN1YiI6IjEifQ.b6vrRuDLEQ-tfrRYq2v_NG43LiJXo1KMXzgYcZoCN7s';
```
Comment (shoenseiwaso) [inline src/App.vue:22]: Eventually (soon?) we want to open source this. Let's keep API keys and URLs out of the code!

File: src/App.vue [hunk 2]
```diff
@@ -2,32 +2,24 @@
...
     <pre v-if="response">{{ response | prettyJSON }}</pre>
   </div>
 </template>
 
 <script>
-import axios from 'axios';
-import axiosCookieJarSupport from '@3846masa/axios-cookiejar-support';
-import tough from 'tough-cookie';
+// [NEED TO FILL] The local address where the demo app will be served
+const BASE_URL = 'https://rjlrpdjikze7fn2vscxfuk365y.multibaas.com';
```
Comment (shoenseiwaso) [inline src/App.vue:19]: See the note about API keys - it applies to specific URLs as well!

**Review by shoenseiwaso — COMMENTED** [review]
Nice updates! Will leave the bulk of the review to @n0thingness (and/or others), but one note about API keys and unique URLs.

File: src/App.vue (hunk 1 above)
Comment (daenamkim) [reply src/App.vue]: Yep, I have already fixed https://github.com/curvegrid/vue-dapp-sample/pull/12/commits/5a1f9ab478247340a7d0cfb287eb729caf7a05cf I just forgot to push 😅

File: src/App.vue (hunk 2 above)
Comment (daenamkim) [reply src/App.vue]: Same with above

File: src/App.vue (hunk 1 above)
Comment (ann-kilzer) [reply src/App.vue]: Friendly reminder it's already public 😄

Comment (shoenseiwaso) [reply src/App.vue]: Indeed!

Comment (daenamkim) [reply src/App.vue]: Yeah that is fine because I have already removed that instance and api key. 😅

**Comment by daenamkim** [comment]
@n0thingness Please take a look at when you are available, no need to rush. :)

**Review by william — APPROVED** [review]
LGTM 🚀 

@daenamkim , please do a review on my additional changes before merging. In the end I pulled out the Mock3 config. It is supremely dangerous to encourage users to store private Ethereum keys in plain text. Especially in the source code. And even more so because in our current setup they will be packaged with the production build.

I think we can revisit this in the future but there has to be a way to make it:
- Clear that it is a test account subject to public exposure and control
- Configurable in a way that it is in a gitignore config file and not in the actual UI code
- Injected by testcafe so that it is not in the actual build (thanks Pierre for reminder about this functionality https://devexpress.github.io/testcafe/documentation/guides/advanced-guides/inject-client-scripts.html)
- (Alternatively we could use a dedicated curvenet server, HSM enabled server, or mock out MultiBaas itself for the tests)

This will be the first interaction with MultiBaas for many new users, perhaps new to blockchain, and we do not want their first action to be exporting private keys insecurely 😉 

Anyway, it was a good idea to add tests 👍  I kept the get balance test and marked the mint test as 'skip'.

Thanks for upgrading us to the API from the username / password and thanks for being patient about the review! 🍦

**Pull request merged**